mobile/src/employee-details/employee-details-screen.tsx

```diff
@@ -2,42 +2,124 @@ import React, { Component } from 'react';
 import { Department } from '../reducers/organization/department.model';
 import { AppState } from '../reducers/app.reducer';
 import { connect } from 'react-redux';
-import { SafeAreaView, ViewStyle } from 'react-native';
-import { profileScreenStyles } from '../profile/styles';
+import { LayoutChangeEvent, RefreshControl, SafeAreaView, View, ViewStyle } from 'react-native';
 import { EmployeeDetails } from './employee-details';
-import { layoutStylesForEmployeeDetailsScreen } from './styles';
-import { NavigationScreenProps } from 'react-navigation';
+import { employeeDetailsStyles } from './styles';
+import { NavigationScreenProps, ScrollView } from 'react-navigation';
 import { LoadingView } from '../navigation/loading';
+import Style from '../layout/style';
+import { layoutStyles } from '../profile/styles';
+import { Action, Dispatch } from 'redux';
+import { refresh } from '../reducers/refresh/refresh.action';
+import { Set } from 'immutable';
 
 //============================================================================
 interface EmployeeDetailsProps {
-    departments: Department[];
+    departments: Set<Department>;
 }
 
 //============================================================================
-class EmployeeDetailsScreenImpl extends Component<EmployeeDetailsProps & NavigationScreenProps> {
+interface EmployeeDetailsDispatchProps {
+    refresh: () => void;
+}
+
+//============================================================================
+interface EmployeeDetailsState {
+    width: number;
+    height: number;
+}
+
+//============================================================================
+class EmployeeDetailsScreenImpl extends Component<EmployeeDetailsProps & EmployeeDetailsDispatchProps & NavigationScreenProps, EmployeeDetailsState> {
+
+    //----------------------------------------------------------------------------
+    public constructor(props: Readonly<EmployeeDetailsProps & EmployeeDetailsDispatchProps & NavigationScreenProps>) {
+        super(props);
+
+        this.state = {
+            width: 0,
+            height: 0,
+        };
+    }
+
+    //----------------------------------------------------------------------------
+    public shouldComponentUpdate(nextProps: EmployeeDetailsProps & EmployeeDetailsDispatchProps & NavigationScreenProps, nextState: EmployeeDetailsState) {
+
+        if (this.state.width !== nextState.width || this.state.height !== nextState.height) {
+            return true;
+        }
+
+        const departments = this.props.departments;
+        const nextDepartments = nextProps.departments;
```

Comment:
probably not relevant but, if nextDepartments becomes null or undefined suddenly, the component will not rerender as it returns true.

Reply:
a86721f8d043aea5fdf571a64c75b7d7cd304bc7